File: Tablas de frecuencia.py
datos = []
n = int(input("¿Cuantos datos va a ingresar?: "))

for i in range(n):
    dato = int(input(f"Ingrese el dato {i+1}: "))
    datos.append(dato)

valores_unicos = sorted(list(set(datos)))  
frecuencia_absoluta = []
frecuencia_acumulada = []
frecuencia_relativa = []
frecuencia_relativa_acumulada = []
porcentajes = []

for valor in valores_unicos:
    frecuencia_absoluta.append(datos.count(valor))

total_datos = sum(frecuencia_absoluta)
acumulada_abs = 0
acumulada_rel = 0

for i in range(len(valores_unicos)):
    acumulada_abs += frecuencia_absoluta[i]
    frecuencia_acumulada.append(acumulada_abs)

    frec_relativa = frecuencia_absoluta[i] / total_datos
    frecuencia_relativa.append(frec_relativa)

    acumulada_rel += frec_relativa
    frecuencia_relativa_acumulada.append(acumulada_rel)

    porcentaje = frec_relativa * 100
    porcentajes.append(porcentaje)

print(f"{'Dato':^10} {'FA':^10} {'FA Acum':^10} {'FR':^10} {'FR Acum':^10} {'%':^10}")
for i in range(len(valores_unicos)):
    print(f"{valores_unicos[i]:^10} {frecuencia_absoluta[i]:^10} {frecuencia_acumulada[i]:^10} {frecuencia_relativa[i]:^10.4f} {frecuencia_relativa_acumulada[i]:^10.4f} {porcentajes[i]:^10.2f}")

print(f"{'Total:':^10} {sum(frecuencia_absoluta):^10} {'':^10} {sum(frecuencia_relativa):^10.4f} {'':^10} {sum(porcentajes):^10.2f}")

File: Tablas de frecuencia.cpp
#include <iostream>
#include <vector>
#include <algorithm> 
#include <numeric>   
#include <iomanip>   

using namespace std;

int main() {
    int n;
    
    cout << "¿Cuántos datos va a ingresar?: ";
    cin >> n;

    vector<int> datos(n);

    for (int i = 0; i < n; ++i) {
        cout << "Ingrese el dato " << i + 1 << ": ";
        cin >> datos[i];
    }

    vector<int> valores_unicos = datos;
    sort(valores_unicos.begin(), valores_unicos.end());
    valores_unicos.erase(unique(valores_unicos.begin(), valores_unicos.end()), valores_unicos.end());

    vector<int> frecuencia_absoluta;
    vector<int> frecuencia_acumulada;
    vector<double> frecuencia_relativa;
    vector<double> frecuencia_relativa_acumulada;
    vector<double> porcentajes;

    int total_datos = datos.size();
    int acumulada_abs = 0;
    double acumulada_rel = 0.0;

    for (int valor : valores_unicos) {
        int count = count_if(datos.begin(), datos.end(), [valor](int x) { return x == valor; });
        frecuencia_absoluta.push_back(count);

        acumulada_abs += count;
        frecuencia_acumulada.push_back(acumulada_abs);

        double frec_relativa = static_cast<double>(count) / total_datos;
        frecuencia_relativa.push_back(frec_relativa);

        acumulada_rel += frec_relativa;
        frecuencia_relativa_acumulada.push_back(acumulada_rel);

        porcentajes.push_back(frec_relativa * 100);
    }

    cout << setw(10) << "Dato" 
         << setw(10) << "FA" 
         << setw(15) << "FA Acum" 
         << setw(10) << "FR" 
         << setw(15) << "FR Acum" 
         << setw(10) << "%" << endl;

    for (size_t i = 0; i < valores_unicos.size(); ++i) {
        cout << setw(10) << valores_unicos[i] 
             << setw(10) << frecuencia_absoluta[i] 
             << setw(15) << frecuencia_acumulada[i] 
             << fixed << setprecision(4) << setw(10) << frecuencia_relativa[i] 
             << fixed << setprecision(4) << setw(15) << frecuencia_relativa_acumulada[i] 
             << fixed << setprecision(2) << setw(10) << porcentajes[i] << "%" << endl;
    }

    cout << setw(10) << "Total:"
         << setw(10) << accumulate(frecuencia_absoluta.begin(), frecuencia_absoluta.end(), 0)
         << setw(15) << " " 
         << fixed << setprecision(4) << setw(10) << accumulate(frecuencia_relativa.begin(), frecuencia_relativa.end(), 0.0)
         << setw(15) << " "
         << fixed << setprecision(2) << setw(10) << accumulate(porcentajes.begin(), porcentajes.end(), 0.0) << "%" << endl;

    return 0;
}